.pipelines/templates/template-prod-e2e-steps.yml

```diff
@@ -1,29 +1,55 @@
 parameters:
+  aroVersionStorageAccount: ''
+  azureDevOpsE2EJSONSPN: ''
   location: ''
+  rpMode: ''
   subscription: ''
-  azureDevOpsE2EJSONSPN: ''
 
 steps:
-# TODO(mj) we need to checkout code to run tooling like kubeconfig. This should go away once we refactor everything into golang.
-- template: ./template-checkout.yml
-- template: ./template-az-cli-login.yml
-  parameters:
-    azureDevOpsJSONSPN: ${{ parameters.azureDevOpsE2EJSONSPN }}
 - script: |
     set -e
     export LOCATION=${{ parameters.location }}
     export AZURE_SUBSCRIPTION_ID=${{ parameters.subscription }}
 
-    trap 'rm -f devops-spn.json' EXIT
+    trap 'set +e; rm -f devops-spn.json ; for c in $(docker ps -aq); do docker rm -f $c; done; docker image prune -af; rm -rf ~/.docker/config.json' EXIT
     base64 -d >devops-spn.json <<<${{ parameters.azureDevOpsE2EJSONSPN }}
     export AZURE_CLIENT_ID=$(jq -r .clientId <devops-spn.json)
     export AZURE_CLIENT_SECRET=$(jq -r .clientSecret <devops-spn.json)
     export AZURE_TENANT_ID=$(jq -r .tenantId <devops-spn.json)
 
+
+    # TODO: read RP config and derive $IMAGE from it
```

Comment:
More comment: this would also need the AZURE_CLIENT_ID identity to be able to pull from non-INT ACR.